src/client/pfs.go

```diff
@@ -0,0 +1,518 @@
+package client
+
+import (
+	"io"
+	"math"
+
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+	"go.pedge.io/proto/stream"
+	"golang.org/x/net/context"
+)
+
+func NewRepo(repoName string) *pfs.Repo {
+	return &pfs.Repo{Name: repoName}
+}
+
+func NewCommit(repoName string, commitID string) *pfs.Commit {
+	return &pfs.Commit{
+		Repo: NewRepo(repoName),
+		ID:   commitID,
+	}
+}
+
+func NewFile(repoName string, commitID string, path string) *pfs.File {
+	return &pfs.File{
+		Commit: NewCommit(repoName, commitID),
+		Path:   path,
+	}
+}
+
+func NewBlock(hash string) *pfs.Block {
+	return &pfs.Block{
+		Hash: hash,
+	}
+}
+
+func NewDiff(repoName string, commitID string, shard uint64) *pfs.Diff {
+	return &pfs.Diff{
+		Commit: NewCommit(repoName, commitID),
+		Shard:  shard,
+	}
+}
+
+const (
+	NONE  = pfs.CommitType_COMMIT_TYPE_NONE
```

Comment:
I went back and forth on these myself, how about `CommitTypeNone` and for the other constants `Map` and `Reduce` we'll call them `InputTypeMap` and `InputTypeReduce`.